client/client/main.go

```diff
@@ -40,91 +40,100 @@ func addAllFiles(cli *client.Client, lastIndexed time.Time) filepath.WalkFunc {
+			}
 		}
 		time.Sleep(time.Second * 60)
 	}
 }
 
-// fetchMasterSecret fetches the master secret from the client directory.  If
+// fetchMasterSecrets fetches the master secrets from the client directories.  If
 // the master secret is not present, a new one is generated and written to the
-// directory.  An error is returned if the there is an issue accessing the
-// master secret or the master secret is of the wrong length.
-func fetchMasterSecret() ([]byte, error) {
-	var masterSecret []byte
-
-	f, err := os.OpenFile(filepath.Join(*clientDirectory, ".search_kbfs_secret"), os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
-
-	if err == nil {
-		defer f.Close()
-		// Generate a random master secret and write it to file
-		masterSecret = make([]byte, *lenMS)
-		if _, err := rand.Read(masterSecret); err != nil {
-			return nil, err
-		}
+// directory.  An error is returned if there is an issue accessing the master
+// secret or the master secret is of the wrong length.
+func fetchMasterSecrets(clientDirs []string) ([][]byte, error) {
+	masterSecrets := make([][]byte, len(clientDirs))
 
-		_, err = f.Write(masterSecret)
-		if err != nil {
-			return nil, err
-		}
-	} else if os.IsExist(err) {
-		// Read the master secret from file
-		masterSecret, err = ioutil.ReadFile(filepath.Join(*clientDirectory, ".search_kbfs_secret"))
-		if err != nil {
+	for i, clientDir := range clientDirs {
+
+		f, err := os.OpenFile(filepath.Join(clientDir, ".search_kbfs_secret"), os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+
+		if err == nil {
+			defer f.Close()
+			// Generate a random master secret and write it to file
+			masterSecrets[i] = make([]byte, *lenMS)
+			if _, err := rand.Read(masterSecrets[i]); err != nil {
+				return nil, err
+			}
+
+			_, err = f.Write(masterSecrets[i])
+			if err != nil {
+				return nil, err
+			}
+		} else if os.IsExist(err) {
+			// Read the master secret from file
+			masterSecrets[i], err = ioutil.ReadFile(filepath.Join(clientDir, ".search_kbfs_secret"))
+			if err != nil {
+				return nil, err
+			}
+			if len(masterSecrets[i]) != *lenMS {
+				return nil, errors.New("Invalid master secret length")
+			}
+		} else {
 			return nil, err
 		}
-		if len(masterSecret) != *lenMS {
-			return nil, errors.New("Invalid master secret length")
-		}
-	} else {
-		return nil, err
 	}
-	return masterSecret, nil
+	return masterSecrets, nil
 }
 
-// performSearchWord searched for the word `keyword` on `cli`, and print out the
-// results.
-func performSearchWord(cli *client.Client, keyword string) {
-	filenames, err := cli.SearchWordStrict(keyword)
-	if err != nil {
-		fmt.Printf("Error when searching word %s: %s", keyword, err)
-		return
+// performSearchWord searches for the word `keyword` on `cli`, and prints out
+// the results.
+func performSearchWord(cli *client.Client, clientDirs []string, keyword string) {
+	var allFiles []string
+	for _, clientDir := range clientDirs {
+		filenames, err := cli.SearchWordStrict(clientDir, keyword)
```

Comment:
Maybe add a TODO here that we can parallelize these calls for better performance.